.golangci.yml

```diff
@@ -99,6 +99,8 @@ linters-settings:
   gocritic:
     enabled-tags:
       - "experimental"
+  gocyclo:
+    min-complexity: 31
```

Comment:
Not relevant to this PR, but I'd be +1 to skip running gocyclo entirely since these kinds of lint errors are more annoying than helpful IMO if they only motivate us to keep raising the ceiling like this.

Reply:
I agree, if you want I can disable it. I can also export code to function and keep it at 30 if you wish.

Reply:
I think this minimally-impacting change here is fine as-is for this PR. Disabling that linter entirely may warrant wider discussion that might fit best in a separate PR.

Reply:
> "Function too complex" in general I think is subjective enough that human eyes seem like a better tool.

What specific function is causing the error here? I agree that linters are not always right, but that doesn't mean they're not helpful. Sometimes it's a useful nudge to reduce tech debt. For each case that the linter complains about we (PR author and reviewers) should carefully consider the function causing the issue and assess whether:
1) the linter is right, the function _is_ getting too complex: refactor it into smaller, testable functions  
2) the linter missed something, there is a good reason for the function length/complexity: add a `//nolint` [comment](https://golangci-lint.run/usage/false-positives/#nolint-directive) to the function that disables the linter for that specific function with a justification

I don't think raising the overall ceiling to "fix" the issue is the right approach. The fact that we're having to add this new configuration in `.golangci.yml` means we've never lifted the ceiling in the several years since the linter was introduced (that's a good thing). Introducing a specific configuration now would be a regression IMO. 

I'm also -1 on disabling the linter entirely in case that wasn't clear, but that should definitely be it's own discussion :)

Reply:
> What specific function is causing the error here? 

It's `Parameters` function, where all fields are copied. The ones with pointers require additional checks that sum up.

If you want I can disable linting for this function or extract code to separate method (but then it's a bit inconsistent with the rest of whole function).

Reply:
I'm +1 to ignoring gocyclo for this particular function or raising the ceiling in a more localized way if that's possible. Extracting code from there into separate functions hasn't reduced complexity or increased reasability/testability IMO.

The way this function is structured currently inherently requires lots of `if <field is nil> { <set field> }` blocks that count against gocyclo but don't seem to me like spaghetti code like I think the linter is trying to point out.

Reply:
Let's disable gocyclo for this particular function 👍

Reply:
Done.